desktop/lib.go

```diff
@@ -255,6 +253,27 @@ func getUserData() (*protos.User, error) {
 	return user, nil
 }
 
+// this method is reposible for checking if the user has updated plan or bought plans
+//
+//export hasPlanUpdatedOrBuy
+func hasPlanUpdatedOrBuy() *C.char {
+	//Get the cached user data
+	uc := userConfig(a.Settings())
+	cahcheUserData, isOldFound := app.GetUserDataFast(context.Background(), uc.GetUserID())
+	//Get latest user data
+	resp, err := proClient.UserData(context.Background())
+	if err != nil {
+		return sendError(err)
+	}
+	if isOldFound {
+		if cahcheUserData.Expiration < resp.User.Expiration {
+			// New data has a later expiration
+			return C.CString(string("true"))
```

Comment:
Nice, I think this is a simple, effective way to check if a user became Pro or renewed their subscription!

Reply:
Thanks!